The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26638616593/job/78505428219) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/78505428219) [(plain)](https://github.com/rust-lang/rust/commit/78eb52bd14711d18d8a8bdca27c91604b2da9226/checks/78505428219/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [assembly] tests/assembly-llvm/compiletest-self-test/use-minicore-no-run.rs ... ok
test [assembly] tests/assembly-llvm/cstring-merging.rs ... ok
test [assembly] tests/assembly-llvm/cmse.rs#hard ... ok
test [assembly] tests/assembly-llvm/cmse.rs#soft ... ok
test [assembly] tests/assembly-llvm/c-variadic/xtensa.rs#XTENSA ... FAILED
test [assembly] tests/assembly-llvm/emit-intel-att-syntax.rs#att ... ignored, only executed when the architecture is x86_64
test [assembly] tests/assembly-llvm/emit-intel-att-syntax.rs#intel ... ignored, only executed when the architecture is x86_64
test [assembly] tests/assembly-llvm/force-target-feature.rs ... ignored, only executed when the architecture is x86_64
test [assembly] tests/assembly-llvm/is_aligned.rs#opt-size ... ignored, only executed when the architecture is x86_64
test [assembly] tests/assembly-llvm/is_aligned.rs#opt-speed ... ignored, only executed when the architecture is x86_64
---
test [assembly] tests/assembly-llvm/x86_64-windows-i128-abi.rs#softfloat ... ok

failures:

---- [assembly] tests/assembly-llvm/c-variadic/xtensa.rs#XTENSA stdout ----
------FileCheck stdout------------------------------

------FileCheck stderr------------------------------
/checkout/tests/assembly-llvm/c-variadic/xtensa.rs:50:13: error: XTENSA: expected string not found in input
 // XTENSA: maxu [[F64_STACK_OFF:a[0-9]+]], [[F64_OFF]], {{a[0-9]+}}
            ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:12:16: note: scanning from here
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:12:16: note: with "F64_OFF" equal to "a9"
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:16:2: note: possible intended match here
 add a9, a10, a9
 ^
/checkout/tests/assembly-llvm/c-variadic/xtensa.rs:79:13: error: XTENSA: expected string not found in input
 // XTENSA: maxu [[I32_STACK_OFF:a[0-9]+]], [[I32_OFF]], {{a[0-9]+}}
            ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:47:16: note: scanning from here
 addi a8, a9, 4
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:47:16: note: with "I32_OFF" equal to "a9"
 addi a8, a9, 4
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:51:2: note: possible intended match here
 add a9, a10, a9
 ^
/checkout/tests/assembly-llvm/c-variadic/xtensa.rs:106:13: error: XTENSA: expected string not found in input
 // XTENSA: maxu [[I64_STACK_OFF:a[0-9]+]], [[I64_OFF]], {{a[0-9]+}}
            ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:80:16: note: scanning from here
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:80:16: note: with "I64_OFF" equal to "a9"
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:84:2: note: possible intended match here
 add a9, a10, a9
 ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s
Check file: /checkout/tests/assembly-llvm/c-variadic/xtensa.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
             1:  .file "xtensa.890a17638639935e-cgu.0" 
             2:  .section .text.read_f64,"ax",@progbits 
             3:  .global read_f64 
             4:  .p2align 2 
             5:  .type read_f64,@function 
             6: read_f64: 
             7:  .cfi_startproc 
             8:  l32i a8, a2, 8 
             9:  addi a8, a8, 7 
            10:  movi a9, -8 
            11:  and a9, a8, a9 
            12:  addi a8, a9, 8 
check:50'0                     X error: no match found
check:50'1                       with "F64_OFF" equal to "a9"
            13:  movi a10, 25 
check:50'0      ~~~~~~~~~~~~~~
            14:  bgeu a8, a10, .LBB0_2 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~
            15:  l32i a10, a2, 4 
check:50'0      ~~~~~~~~~~~~~~~~~
            16:  add a9, a10, a9 
check:50'0      ~~~~~~~~~~~~~~~~~
check:50'2       ?                possible intended match
            17:  s32i a8, a2, 8 
check:50'0      ~~~~~~~~~~~~~~~~
            18:  l32i a2, a9, 0 
check:50'0      ~~~~~~~~~~~~~~~~
            19:  l32i a3, a9, 4 
check:50'0      ~~~~~~~~~~~~~~~~
            20:  ret 
check:50'0      ~~~~~
            21: .LBB0_2: 
check:50'0      ~~~~~~~~~
            22:  movi a8, 32 
check:50'0      ~~~~~~~~~~~~~
            23:  bgeu a8, a9, .LBB0_4 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~
            24:  or a8, a9, a9 
check:50'0      ~~~~~~~~~~~~~~~
            25: .LBB0_4: 
check:50'0      ~~~~~~~~~
            26:  l32i a9, a2, 0 
check:50'0      ~~~~~~~~~~~~~~~~
            27:  add a9, a9, a8 
check:50'0      ~~~~~~~~~~~~~~~~
            28:  addi a8, a8, 8 
check:50'0      ~~~~~~~~~~~~~~~~
            29:  s32i a8, a2, 8 
check:50'0      ~~~~~~~~~~~~~~~~
            30:  l32i a2, a9, 0 
check:50'0      ~~~~~~~~~~~~~~~~
            31:  l32i a3, a9, 4 
check:50'0      ~~~~~~~~~~~~~~~~
            32:  ret 
check:50'0      ~~~~~
            33: .Lfunc_end0: 
check:50'0      ~~~~~~~~~~~~~
            34:  .size read_f64, .Lfunc_end0-read_f64 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            35:  .cfi_endproc 
check:50'0      ~~~~~~~~~~~~~~
            36:  
check:50'0      ~
            37:  .section .text.read_i32,"ax",@progbits 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~
            38:  .global read_i32 
            39:  .p2align 2 
            40:  .type read_i32,@function 
            41: read_i32: 
            42:  .cfi_startproc 
            43:  l32i a8, a2, 8 
            44:  addi a8, a8, 3 
            45:  movi a9, -4 
            46:  and a9, a8, a9 
            47:  addi a8, a9, 4 
check:79'0                     X error: no match found
check:79'1                       with "I32_OFF" equal to "a9"
            48:  movi a10, 25 
check:79'0      ~~~~~~~~~~~~~~
            49:  bgeu a8, a10, .LBB1_2 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~~
            50:  l32i a10, a2, 4 
check:79'0      ~~~~~~~~~~~~~~~~~
            51:  add a9, a10, a9 
check:79'0      ~~~~~~~~~~~~~~~~~
check:79'2       ?                possible intended match
            52:  s32i a8, a2, 8 
check:79'0      ~~~~~~~~~~~~~~~~
            53:  l32i a2, a9, 0 
check:79'0      ~~~~~~~~~~~~~~~~
            54:  ret 
check:79'0      ~~~~~
            55: .LBB1_2: 
check:79'0      ~~~~~~~~~
            56:  movi a8, 32 
check:79'0      ~~~~~~~~~~~~~
            57:  bgeu a8, a9, .LBB1_4 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~
            58:  or a8, a9, a9 
check:79'0      ~~~~~~~~~~~~~~~
            59: .LBB1_4: 
check:79'0      ~~~~~~~~~
            60:  l32i a9, a2, 0 
check:79'0      ~~~~~~~~~~~~~~~~
            61:  add a9, a9, a8 
check:79'0      ~~~~~~~~~~~~~~~~
            62:  addi a8, a8, 4 
check:79'0      ~~~~~~~~~~~~~~~~
            63:  s32i a8, a2, 8 
check:79'0      ~~~~~~~~~~~~~~~~
            64:  l32i a2, a9, 0 
check:79'0      ~~~~~~~~~~~~~~~~
            65:  ret 
check:79'0      ~~~~~
            66: .Lfunc_end1: 
check:79'0      ~~~~~~~~~~~~~
            67:  .size read_i32, .Lfunc_end1-read_i32 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            68:  .cfi_endproc 
check:79'0      ~~~~~~~~~~~~~~
            69:  
check:79'0      ~
            70:  .section .text.read_i64,"ax",@progbits 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~~~
            71:  .global read_i64 
            72:  .p2align 2 
            73:  .type read_i64,@function 
            74: read_i64: 
            75:  .cfi_startproc 
            76:  l32i a8, a2, 8 
            77:  addi a8, a8, 7 
            78:  movi a9, -8 
            79:  and a9, a8, a9 
            80:  addi a8, a9, 8 
check:106'0                    X error: no match found
check:106'1                      with "I64_OFF" equal to "a9"
            81:  movi a10, 25 
check:106'0     ~~~~~~~~~~~~~~
            82:  bgeu a8, a10, .LBB2_2 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~
            83:  l32i a10, a2, 4 
check:106'0     ~~~~~~~~~~~~~~~~~
            84:  add a9, a10, a9 
check:106'0     ~~~~~~~~~~~~~~~~~
check:106'2      ?                possible intended match
            85:  s32i a8, a2, 8 
check:106'0     ~~~~~~~~~~~~~~~~
            86:  l32i a2, a9, 0 
check:106'0     ~~~~~~~~~~~~~~~~
            87:  l32i a3, a9, 4 
check:106'0     ~~~~~~~~~~~~~~~~
            88:  ret 
check:106'0     ~~~~~
            89: .LBB2_2: 
check:106'0     ~~~~~~~~~
            90:  movi a8, 32 
check:106'0     ~~~~~~~~~~~~~
            91:  bgeu a8, a9, .LBB2_4 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~
            92:  or a8, a9, a9 
check:106'0     ~~~~~~~~~~~~~~~
            93: .LBB2_4: 
check:106'0     ~~~~~~~~~
            94:  l32i a9, a2, 0 
check:106'0     ~~~~~~~~~~~~~~~~
            95:  add a9, a9, a8 
check:106'0     ~~~~~~~~~~~~~~~~
            96:  addi a8, a8, 8 
check:106'0     ~~~~~~~~~~~~~~~~
            97:  s32i a8, a2, 8 
check:106'0     ~~~~~~~~~~~~~~~~
            98:  l32i a2, a9, 0 
check:106'0     ~~~~~~~~~~~~~~~~
            99:  l32i a3, a9, 4 
check:106'0     ~~~~~~~~~~~~~~~~
           100:  ret 
check:106'0     ~~~~~
           101: .Lfunc_end2: 
check:106'0     ~~~~~~~~~~~~~
           102:  .size read_i64, .Lfunc_end2-read_i64 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           103:  .cfi_endproc 
check:106'0     ~~~~~~~~~~~~~~
           104:  
check:106'0     ~
           105:  .global read_ptr 
check:106'0     ~~~~~~~~~~~~~~~~~~
           106:  .type read_ptr,@function 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
           107: read_ptr = read_i32 
check:106'0     ~~~~~~~~~~~~~~~~~~~~
           108:  .ident "rustc version 1.98.0-nightly (0213b4fd6 2026-05-29)" 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           109:  .section ".note.GNU-stack","",@progbits 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>

------------------------------------------

error in revision `XTENSA`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-21/bin/FileCheck" "--input-file" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s" "/checkout/tests/assembly-llvm/c-variadic/xtensa.rs" "--check-prefix=CHECK" "--check-prefix" "XTENSA" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/assembly-llvm/c-variadic/xtensa.rs:50:13: error: XTENSA: expected string not found in input
 // XTENSA: maxu [[F64_STACK_OFF:a[0-9]+]], [[F64_OFF]], {{a[0-9]+}}
            ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:12:16: note: scanning from here
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:12:16: note: with "F64_OFF" equal to "a9"
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:16:2: note: possible intended match here
 add a9, a10, a9
 ^
/checkout/tests/assembly-llvm/c-variadic/xtensa.rs:79:13: error: XTENSA: expected string not found in input
 // XTENSA: maxu [[I32_STACK_OFF:a[0-9]+]], [[I32_OFF]], {{a[0-9]+}}
            ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:47:16: note: scanning from here
 addi a8, a9, 4
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:47:16: note: with "I32_OFF" equal to "a9"
 addi a8, a9, 4
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:51:2: note: possible intended match here
 add a9, a10, a9
 ^
/checkout/tests/assembly-llvm/c-variadic/xtensa.rs:106:13: error: XTENSA: expected string not found in input
 // XTENSA: maxu [[I64_STACK_OFF:a[0-9]+]], [[I64_OFF]], {{a[0-9]+}}
            ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:80:16: note: scanning from here
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:80:16: note: with "I64_OFF" equal to "a9"
 addi a8, a9, 8
               ^
/checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s:84:2: note: possible intended match here
 add a9, a10, a9
 ^

Input file: /checkout/obj/build/aarch64-unknown-linux-gnu/test/assembly-llvm/c-variadic/xtensa.XTENSA/xtensa.s
Check file: /checkout/tests/assembly-llvm/c-variadic/xtensa.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
             1:  .file "xtensa.890a17638639935e-cgu.0" 
             2:  .section .text.read_f64,"ax",@progbits 
             3:  .global read_f64 
             4:  .p2align 2 
             5:  .type read_f64,@function 
             6: read_f64: 
             7:  .cfi_startproc 
             8:  l32i a8, a2, 8 
             9:  addi a8, a8, 7 
            10:  movi a9, -8 
            11:  and a9, a8, a9 
            12:  addi a8, a9, 8 
check:50'0                     X error: no match found
check:50'1                       with "F64_OFF" equal to "a9"
            13:  movi a10, 25 
check:50'0      ~~~~~~~~~~~~~~
            14:  bgeu a8, a10, .LBB0_2 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~
            15:  l32i a10, a2, 4 
check:50'0      ~~~~~~~~~~~~~~~~~
            16:  add a9, a10, a9 
check:50'0      ~~~~~~~~~~~~~~~~~
check:50'2       ?                possible intended match
            17:  s32i a8, a2, 8 
check:50'0      ~~~~~~~~~~~~~~~~
            18:  l32i a2, a9, 0 
check:50'0      ~~~~~~~~~~~~~~~~
            19:  l32i a3, a9, 4 
check:50'0      ~~~~~~~~~~~~~~~~
            20:  ret 
check:50'0      ~~~~~
            21: .LBB0_2: 
check:50'0      ~~~~~~~~~
            22:  movi a8, 32 
check:50'0      ~~~~~~~~~~~~~
            23:  bgeu a8, a9, .LBB0_4 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~
            24:  or a8, a9, a9 
check:50'0      ~~~~~~~~~~~~~~~
            25: .LBB0_4: 
check:50'0      ~~~~~~~~~
            26:  l32i a9, a2, 0 
check:50'0      ~~~~~~~~~~~~~~~~
            27:  add a9, a9, a8 
check:50'0      ~~~~~~~~~~~~~~~~
            28:  addi a8, a8, 8 
check:50'0      ~~~~~~~~~~~~~~~~
            29:  s32i a8, a2, 8 
check:50'0      ~~~~~~~~~~~~~~~~
            30:  l32i a2, a9, 0 
check:50'0      ~~~~~~~~~~~~~~~~
            31:  l32i a3, a9, 4 
check:50'0      ~~~~~~~~~~~~~~~~
            32:  ret 
check:50'0      ~~~~~
            33: .Lfunc_end0: 
check:50'0      ~~~~~~~~~~~~~
            34:  .size read_f64, .Lfunc_end0-read_f64 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            35:  .cfi_endproc 
check:50'0      ~~~~~~~~~~~~~~
            36:  
check:50'0      ~
            37:  .section .text.read_i32,"ax",@progbits 
check:50'0      ~~~~~~~~~~~~~~~~~~~~~~~~
            38:  .global read_i32 
            39:  .p2align 2 
            40:  .type read_i32,@function 
            41: read_i32: 
            42:  .cfi_startproc 
            43:  l32i a8, a2, 8 
            44:  addi a8, a8, 3 
            45:  movi a9, -4 
            46:  and a9, a8, a9 
            47:  addi a8, a9, 4 
check:79'0                     X error: no match found
check:79'1                       with "I32_OFF" equal to "a9"
            48:  movi a10, 25 
check:79'0      ~~~~~~~~~~~~~~
            49:  bgeu a8, a10, .LBB1_2 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~~
            50:  l32i a10, a2, 4 
check:79'0      ~~~~~~~~~~~~~~~~~
            51:  add a9, a10, a9 
check:79'0      ~~~~~~~~~~~~~~~~~
check:79'2       ?                possible intended match
            52:  s32i a8, a2, 8 
check:79'0      ~~~~~~~~~~~~~~~~
            53:  l32i a2, a9, 0 
check:79'0      ~~~~~~~~~~~~~~~~
            54:  ret 
check:79'0      ~~~~~
            55: .LBB1_2: 
check:79'0      ~~~~~~~~~
            56:  movi a8, 32 
check:79'0      ~~~~~~~~~~~~~
            57:  bgeu a8, a9, .LBB1_4 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~
            58:  or a8, a9, a9 
check:79'0      ~~~~~~~~~~~~~~~
            59: .LBB1_4: 
check:79'0      ~~~~~~~~~
            60:  l32i a9, a2, 0 
check:79'0      ~~~~~~~~~~~~~~~~
            61:  add a9, a9, a8 
check:79'0      ~~~~~~~~~~~~~~~~
            62:  addi a8, a8, 4 
check:79'0      ~~~~~~~~~~~~~~~~
            63:  s32i a8, a2, 8 
check:79'0      ~~~~~~~~~~~~~~~~
            64:  l32i a2, a9, 0 
check:79'0      ~~~~~~~~~~~~~~~~
            65:  ret 
check:79'0      ~~~~~
            66: .Lfunc_end1: 
check:79'0      ~~~~~~~~~~~~~
            67:  .size read_i32, .Lfunc_end1-read_i32 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            68:  .cfi_endproc 
check:79'0      ~~~~~~~~~~~~~~
            69:  
check:79'0      ~
            70:  .section .text.read_i64,"ax",@progbits 
check:79'0      ~~~~~~~~~~~~~~~~~~~~~~~~
            71:  .global read_i64 
            72:  .p2align 2 
            73:  .type read_i64,@function 
            74: read_i64: 
            75:  .cfi_startproc 
            76:  l32i a8, a2, 8 
            77:  addi a8, a8, 7 
            78:  movi a9, -8 
            79:  and a9, a8, a9 
            80:  addi a8, a9, 8 
check:106'0                    X error: no match found
check:106'1                      with "I64_OFF" equal to "a9"
            81:  movi a10, 25 
check:106'0     ~~~~~~~~~~~~~~
            82:  bgeu a8, a10, .LBB2_2 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~
            83:  l32i a10, a2, 4 
check:106'0     ~~~~~~~~~~~~~~~~~
            84:  add a9, a10, a9 
check:106'0     ~~~~~~~~~~~~~~~~~
check:106'2      ?                possible intended match
            85:  s32i a8, a2, 8 
check:106'0     ~~~~~~~~~~~~~~~~
            86:  l32i a2, a9, 0 
check:106'0     ~~~~~~~~~~~~~~~~
            87:  l32i a3, a9, 4 
check:106'0     ~~~~~~~~~~~~~~~~
            88:  ret 
check:106'0     ~~~~~
            89: .LBB2_2: 
check:106'0     ~~~~~~~~~
            90:  movi a8, 32 
check:106'0     ~~~~~~~~~~~~~
            91:  bgeu a8, a9, .LBB2_4 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~
            92:  or a8, a9, a9 
check:106'0     ~~~~~~~~~~~~~~~
            93: .LBB2_4: 
check:106'0     ~~~~~~~~~
            94:  l32i a9, a2, 0 
check:106'0     ~~~~~~~~~~~~~~~~
            95:  add a9, a9, a8 
check:106'0     ~~~~~~~~~~~~~~~~
            96:  addi a8, a8, 8 
check:106'0     ~~~~~~~~~~~~~~~~
            97:  s32i a8, a2, 8 
check:106'0     ~~~~~~~~~~~~~~~~
            98:  l32i a2, a9, 0 
check:106'0     ~~~~~~~~~~~~~~~~
            99:  l32i a3, a9, 4 
check:106'0     ~~~~~~~~~~~~~~~~
           100:  ret 
check:106'0     ~~~~~
           101: .Lfunc_end2: 
check:106'0     ~~~~~~~~~~~~~
           102:  .size read_i64, .Lfunc_end2-read_i64 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           103:  .cfi_endproc 
check:106'0     ~~~~~~~~~~~~~~
           104:  
check:106'0     ~
           105:  .global read_ptr 
check:106'0     ~~~~~~~~~~~~~~~~~~
           106:  .type read_ptr,@function 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
           107: read_ptr = read_i32 
check:106'0     ~~~~~~~~~~~~~~~~~~~~
           108:  .ident "rustc version 1.98.0-nightly (0213b4fd6 2026-05-29)" 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           109:  .section ".note.GNU-stack","",@progbits 
check:106'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>
------------------------------------------

---- [assembly] tests/assembly-llvm/c-variadic/xtensa.rs#XTENSA stdout end ----

```

</details>

--- a/tests/assembly-llvm/c-variadic/xtensa.rs
+++ b/tests/assembly-llvm/c-variadic/xtensa.rs
@@ -1,5 +1,6 @@
 //@ add-minicore
 //@ assembly-output: emit-asm
+//@ min-llvm-version: 22
 //
 //@ revisions: XTENSA
 //@ [XTENSA] compile-flags: -Copt-level=3 --target xtensa-esp32-none-elf
@@ -42,27 +43,30 @@ pub const unsafe fn va_arg<T: VaArgSafe>(ap: &mut VaList<'_>) -> T;
 unsafe extern "C" fn read_f64(ap: &mut VaList<'_>) -> f64 {
     // CHECK-LABEL: read_f64
     //
-    // XTENSA: l32i{{(\.n)?}} [[F64_NDX:a[0-9]+]], a2, 8
-    // XTENSA: addi{{(\.n)?}} [[F64_ALIGNED_TMP:a[0-9]+]], [[F64_NDX]], 7
-    // XTENSA: movi{{(\.n)?}} [[F64_ALIGN_MASK:a[0-9]+]], -8
-    // XTENSA: and [[F64_OFF:a[0-9]+]], [[F64_ALIGNED_TMP]], [[F64_ALIGN_MASK]]
-    // XTENSA: addi{{(\.n)?}} [[F64_NEXT:a[0-9]+]], [[F64_OFF]], 8
-    // XTENSA: maxu [[F64_STACK_OFF:a[0-9]+]], [[F64_OFF]], {{a[0-9]+}}
-    // XTENSA: bltu [[F64_NEXT]], {{a[0-9]+}}, [[F64_REGSAVE:.LBB[0-9]+_[0-9]+]]
-    // XTENSA: addi{{(\.n)?}} [[F64_UPDATED_NDX:a[0-9]+]], [[F64_STACK_OFF]], 8
-    // XTENSA: [[F64_UPDATE:.LBB[0-9]+_[0-9]+]]:
-    // XTENSA-NEXT: s32i{{(\.n)?}} {{a[0-9]+}}, a2, 8
-    // XTENSA: bltu [[F64_NEXT]], {{a[0-9]+}}, [[F64_REGSAVE_LOAD:.LBB[0-9]+_[0-9]+]]
-    // XTENSA: l32i{{(\.n)?}} [[F64_STACK:a[0-9]+]], a2, 0
-    // XTENSA: add{{(\.n)?}} [[F64_PTR:a[0-9]+]], [[F64_STACK]], [[F64_STACK_OFF]]
-    // XTENSA: l32i{{(\.n)?}} a2, [[F64_PTR]], 0
-    // XTENSA-NEXT: l32i{{(\.n)?}} a3, [[F64_PTR]], 4
+    // XTENSA: l32i a8, a2, 8
+    // XTENSA-NEXT: addi a8, a8, 7
+    // XTENSA-NEXT: movi a9, -8
+    // XTENSA-NEXT: and a8, a8, a9
+    // XTENSA-NEXT: addi a10, a8, 8
+    // XTENSA-NEXT: movi a9, 32
+    // XTENSA-NEXT: maxu a9, a8, a9
+    // XTENSA-NEXT: movi a11, 25
+    // XTENSA-NEXT: or a12, a10, a10
+    // XTENSA-NEXT: bltu a10, a11, .LBB0_2
+    // XTENSA-NEXT: addi a12, a9, 8
+    // XTENSA-NEXT: .LBB0_2:
+    // XTENSA-NEXT: s32i a12, a2, 8
+    // XTENSA-NEXT: bltu a10, a11, .LBB0_4
+    // XTENSA-NEXT: l32i a8, a2, 0
+    // XTENSA-NEXT: add a8, a8, a9
+    // XTENSA-NEXT: l32i a2, a8, 0
+    // XTENSA-NEXT: l32i a3, a8, 4
     // XTENSA-NEXT: retw.n
-    // XTENSA: [[F64_REGSAVE_LOAD]]:
-    // XTENSA-NEXT: l32i{{(\.n)?}} [[F64_REGSAVE_AREA:a[0-9]+]], a2, 4
-    // XTENSA-NEXT: add{{(\.n)?}} [[F64_REGSAVE_PTR:a[0-9]+]], [[F64_REGSAVE_AREA]], [[F64_OFF]]
-    // XTENSA-NEXT: l32i{{(\.n)?}} a2, [[F64_REGSAVE_PTR]], 0
-    // XTENSA-NEXT: l32i{{(\.n)?}} a3, [[F64_REGSAVE_PTR]], 4
+    // XTENSA-NEXT: .LBB0_4:
+    // XTENSA-NEXT: l32i a9, a2, 4
+    // XTENSA-NEXT: add a8, a9, a8
+    // XTENSA-NEXT: l32i a2, a8, 0
+    // XTENSA-NEXT: l32i a3, a8, 4
     // XTENSA-NEXT: retw.n
     va_arg(ap)
 }
@@ -71,25 +75,28 @@ unsafe extern "C" fn read_f64(ap: &mut VaList<'_>) -> f64 {
 unsafe extern "C" fn read_i32(ap: &mut VaList<'_>) -> i32 {
     // CHECK-LABEL: read_i32
     //
-    // XTENSA: l32i{{(\.n)?}} [[I32_NDX:a[0-9]+]], a2, 8
-    // XTENSA: addi{{(\.n)?}} [[I32_ALIGNED_TMP:a[0-9]+]], [[I32_NDX]], 3
-    // XTENSA: movi{{(\.n)?}} [[I32_ALIGN_MASK:a[0-9]+]], -4
-    // XTENSA: and [[I32_OFF:a[0-9]+]], [[I32_ALIGNED_TMP]], [[I32_ALIGN_MASK]]
-    // XTENSA: addi{{(\.n)?}} [[I32_NEXT:a[0-9]+]], [[I32_OFF]], 4
-    // XTENSA: maxu [[I32_STACK_OFF:a[0-9]+]], [[I32_OFF]], {{a[0-9]+}}
-    // XTENSA: bltu [[I32_NEXT]], {{a[0-9]+}}, [[I32_REGSAVE:.LBB[0-9]+_[0-9]+]]
-    // XTENSA: addi{{(\.n)?}} [[I32_UPDATED_NDX:a[0-9]+]], [[I32_STACK_OFF]], 4
-    // XTENSA: [[I32_UPDATE:.LBB[0-9]+_[0-9]+]]:
-    // XTENSA-NEXT: s32i{{(\.n)?}} {{a[0-9]+}}, a2, 8
-    // XTENSA: bltu [[I32_NEXT]], {{a[0-9]+}}, [[I32_REGSAVE_LOAD:.LBB[0-9]+_[0-9]+]]
-    // XTENSA: l32i{{(\.n)?}} [[I32_STACK:a[0-9]+]], a2, 0
-    // XTENSA: add{{(\.n)?}} [[I32_PTR:a[0-9]+]], [[I32_STACK]], [[I32_STACK_OFF]]
-    // XTENSA: l32i{{(\.n)?}} a2, [[I32_PTR]], 0
+    // XTENSA: l32i a8, a2, 8
+    // XTENSA-NEXT: addi a8, a8, 3
+    // XTENSA-NEXT: movi a9, -4
+    // XTENSA-NEXT: and a8, a8, a9
+    // XTENSA-NEXT: addi a10, a8, 4
+    // XTENSA-NEXT: movi a9, 32
+    // XTENSA-NEXT: maxu a9, a8, a9
+    // XTENSA-NEXT: movi a11, 25
+    // XTENSA-NEXT: or a12, a10, a10
+    // XTENSA-NEXT: bltu a10, a11, .LBB1_2
+    // XTENSA-NEXT: addi a12, a9, 4
+    // XTENSA-NEXT: .LBB1_2:
+    // XTENSA-NEXT: s32i a12, a2, 8
+    // XTENSA-NEXT: bltu a10, a11, .LBB1_4
+    // XTENSA-NEXT: l32i a8, a2, 0
+    // XTENSA-NEXT: add a8, a8, a9
+    // XTENSA-NEXT: l32i a2, a8, 0
     // XTENSA-NEXT: retw.n
-    // XTENSA: [[I32_REGSAVE_LOAD]]:
-    // XTENSA-NEXT: l32i{{(\.n)?}} [[I32_REGSAVE_AREA:a[0-9]+]], a2, 4
-    // XTENSA-NEXT: add{{(\.n)?}} [[I32_REGSAVE_PTR:a[0-9]+]], [[I32_REGSAVE_AREA]], [[I32_OFF]]
-    // XTENSA-NEXT: l32i{{(\.n)?}} a2, [[I32_REGSAVE_PTR]], 0
+    // XTENSA-NEXT: .LBB1_4:
+    // XTENSA-NEXT: l32i a9, a2, 4
+    // XTENSA-NEXT: add a8, a9, a8
+    // XTENSA-NEXT: l32i a2, a8, 0
     // XTENSA-NEXT: retw.n
     va_arg(ap)
 }
@@ -98,27 +105,30 @@ unsafe extern "C" fn read_i32(ap: &mut VaList<'_>) -> i32 {
 unsafe extern "C" fn read_i64(ap: &mut VaList<'_>) -> i64 {
     // CHECK-LABEL: read_i64
     //
-    // XTENSA: l32i{{(\.n)?}} [[I64_NDX:a[0-9]+]], a2, 8
-    // XTENSA: addi{{(\.n)?}} [[I64_ALIGNED_TMP:a[0-9]+]], [[I64_NDX]], 7
-    // XTENSA: movi{{(\.n)?}} [[I64_ALIGN_MASK:a[0-9]+]], -8
-    // XTENSA: and [[I64_OFF:a[0-9]+]], [[I64_ALIGNED_TMP]], [[I64_ALIGN_MASK]]
-    // XTENSA: addi{{(\.n)?}} [[I64_NEXT:a[0-9]+]], [[I64_OFF]], 8
-    // XTENSA: maxu [[I64_STACK_OFF:a[0-9]+]], [[I64_OFF]], {{a[0-9]+}}
-    // XTENSA: bltu [[I64_NEXT]], {{a[0-9]+}}, [[I64_REGSAVE:.LBB[0-9]+_[0-9]+]]
-    // XTENSA: addi{{(\.n)?}} [[I64_UPDATED_NDX:a[0-9]+]], [[I64_STACK_OFF]], 8
-    // XTENSA: [[I64_UPDATE:.LBB[0-9]+_[0-9]+]]:
-    // XTENSA-NEXT: s32i{{(\.n)?}} {{a[0-9]+}}, a2, 8
-    // XTENSA: bltu [[I64_NEXT]], {{a[0-9]+}}, [[I64_REGSAVE_LOAD:.LBB[0-9]+_[0-9]+]]
-    // XTENSA: l32i{{(\.n)?}} [[I64_STACK:a[0-9]+]], a2, 0
-    // XTENSA: add{{(\.n)?}} [[I64_PTR:a[0-9]+]], [[I64_STACK]], [[I64_STACK_OFF]]
-    // XTENSA: l32i{{(\.n)?}} a2, [[I64_PTR]], 0
-    // XTENSA-NEXT: l32i{{(\.n)?}} a3, [[I64_PTR]], 4
+    // XTENSA: l32i a8, a2, 8
+    // XTENSA-NEXT: addi a8, a8, 7
+    // XTENSA-NEXT: movi a9, -8
+    // XTENSA-NEXT: and a8, a8, a9
+    // XTENSA-NEXT: addi a10, a8, 8
+    // XTENSA-NEXT: movi a9, 32
+    // XTENSA-NEXT: maxu a9, a8, a9
+    // XTENSA-NEXT: movi a11, 25
+    // XTENSA-NEXT: or a12, a10, a10
+    // XTENSA-NEXT: bltu a10, a11, .LBB2_2
+    // XTENSA-NEXT: addi a12, a9, 8
+    // XTENSA-NEXT: .LBB2_2:
+    // XTENSA-NEXT: s32i a12, a2, 8
+    // XTENSA-NEXT: bltu a10, a11, .LBB2_4
+    // XTENSA-NEXT: l32i a8, a2, 0
+    // XTENSA-NEXT: add a8, a8, a9
+    // XTENSA-NEXT: l32i a2, a8, 0
+    // XTENSA-NEXT: l32i a3, a8, 4
     // XTENSA-NEXT: retw.n
-    // XTENSA: [[I64_REGSAVE_LOAD]]:
-    // XTENSA-NEXT: l32i{{(\.n)?}} [[I64_REGSAVE_AREA:a[0-9]+]], a2, 4
-    // XTENSA-NEXT: add{{(\.n)?}} [[I64_REGSAVE_PTR:a[0-9]+]], [[I64_REGSAVE_AREA]], [[I64_OFF]]
-    // XTENSA-NEXT: l32i{{(\.n)?}} a2, [[I64_REGSAVE_PTR]], 0
-    // XTENSA-NEXT: l32i{{(\.n)?}} a3, [[I64_REGSAVE_PTR]], 4
+    // XTENSA-NEXT: .LBB2_4:
+    // XTENSA-NEXT: l32i a9, a2, 4
+    // XTENSA-NEXT: add a8, a9, a8
+    // XTENSA-NEXT: l32i a2, a8, 0
+    // XTENSA-NEXT: l32i a3, a8, 4
     // XTENSA-NEXT: retw.n
     va_arg(ap)
 }

